### PR TITLE
Balder/gc loadtype set

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@vekterli This is the last large one ....

File: documentapi/src/vespa/documentapi/messagebus/routablerepository.h [hunk 1]
```diff
@@ -40,15 +38,14 @@ class RoutableRepository {
...
-    const LoadTypeSet&   _loadTypes;
 
 public:
     RoutableRepository(const RoutableRepository &) = delete;
     RoutableRepository & operator = (const RoutableRepository &) = delete;
     /**
      * Constructs a new routable repository.
      */
-    explicit RoutableRepository(const LoadTypeSet& loadTypes);
+    explicit RoutableRepository();
```
Comment (vekterli) [inline documentapi/src/vespa/documentapi/messagebus/routablerepository.h:48]: Can remove `explicit` here

File: storage/src/vespa/storage/storageserver/rpc/message_codec_provider.h [hunk 2]
```diff
@@ -34,14 +31,12 @@ class MessageCodecProvider {
     mutable std::shared_mutex _rw_mutex;
     std::shared_ptr<WrappedCodec> _active_codec;
 public:
-    MessageCodecProvider(std::shared_ptr<const document::DocumentTypeRepo> doc_type_repo,
-                         std::shared_ptr<const documentapi::LoadTypeSet> load_type_set);
+    MessageCodecProvider(std::shared_ptr<const document::DocumentTypeRepo> doc_type_repo);
```
Comment (vekterli) [inline storage/src/vespa/storage/storageserver/rpc/message_codec_provider.h:34]: Consider adding `explicit` to `WrappedCodec` and `MessageCodecProvider` ctors

File: storageapi/src/vespa/storageapi/mbusprot/protocolserialization7.h [hunk 3]
```diff
@@ -13,13 +12,10 @@ namespace storage::mbusprot {
  */
 class ProtocolSerialization7 final : public ProtocolSerialization {
     const std::shared_ptr<const document::DocumentTypeRepo> _repo;
-    const documentapi::LoadTypeSet& _load_types;
 public:
-    ProtocolSerialization7(std::shared_ptr<const document::DocumentTypeRepo> repo,
-                           const documentapi::LoadTypeSet& load_types);
+    ProtocolSerialization7(std::shared_ptr<const document::DocumentTypeRepo> repo);
```
Comment (vekterli) [inline storageapi/src/vespa/storageapi/mbusprot/protocolserialization7.h:16]: Consider making `explicit`

File: storageapi/src/vespa/storageapi/mbusprot/storageprotocol.h [hunk 4]
```diff
@@ -15,8 +15,7 @@ class StorageProtocol final : public mbus::IProtocol
 
     static mbus::string NAME;
 
-    StorageProtocol(const std::shared_ptr<const document::DocumentTypeRepo>,
-                    const documentapi::LoadTypeSet& loadTypes);
+    StorageProtocol(const std::shared_ptr<const document::DocumentTypeRepo>);
```
Comment (vekterli) [inline storageapi/src/vespa/storageapi/mbusprot/storageprotocol.h:18]: Consider making `explicit`

**Review by vekterli — APPROVED** [review]
:+1: and there was much rejoicing across the land! 🥳

File: documentapi/src/vespa/documentapi/messagebus/routablerepository.h (hunk 1 above)
Comment (baldersheim) [reply documentapi/src/vespa/documentapi/messagebus/routablerepository.h:48]: Fixed

File: storage/src/vespa/storage/storageserver/rpc/message_codec_provider.h (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/storageserver/rpc/message_codec_provider.h:34]: Fixed

File: storageapi/src/vespa/storageapi/mbusprot/protocolserialization7.h (hunk 3 above)
Comment (baldersheim) [reply storageapi/src/vespa/storageapi/mbusprot/protocolserialization7.h:16]: fixed

File: storageapi/src/vespa/storageapi/mbusprot/storageprotocol.h (hunk 4 above)
Comment (baldersheim) [reply storageapi/src/vespa/storageapi/mbusprot/storageprotocol.h:18]: Fixed

**Pull request merged**